### PR TITLE
Add support for `nav/media-list` container

### LINKED ISSUE #5139: "nav/media-list" (`102`) container
Part of https://github.com/guardian/dotcom-rendering/issues/4720

Example can be found on https://www.theguardian.com/uk/sport in the "Other sports" container.

Please update [FaciaPicker in Frontend](https://github.com/guardian/frontend/blob/main/facia/app/services/dotcomrendering/FaciaPicker.scala) once this container is supported by DCR

### PR DESCRIPTION
**Pull request by AshCorr**
## What does this change?

Adds support for the `nav/media-list` container. I've merged it with the NavList container due to their similar look and feel.

In the DCR version the link is subdued as this is more accessible.

## Why?

More containers!

Closes #5139

## Screenshots

| Before      | After      |
|-------------|------------|
| ![before][] | ![after][] |

[before]: https://user-images.githubusercontent.com/21217225/205668568-a2fd4135-849f-42c1-88b2-46daf3202482.png
[after]: https://user-images.githubusercontent.com/21217225/205668489-43f7fbd7-1713-4289-bd15-63d4b25c04af.png

### CODE REVIEW

**Comment by AshCorr** [comment]
Not stale! but thanks for the reminder Github!

File: dotcom-rendering/src/web/components/MiniCard.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	trail: TrailType;
+	showImage: boolean;
+	containerPalette?: DCRContainerPalette;
+};
+
+export const MiniCard = ({ trail, showImage, containerPalette }: Props) => {
+	const containerOverrides =
+		containerPalette && decideContainerOverrides(containerPalette);
+
+	const image =
```
Comment (AshCorr) [inline dotcom-rendering/src/web/components/MiniCard.tsx:24]: Frontend doesn't use `<picture>` for this either

File: dotcom-rendering/src/web/components/MiniCard.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import { css } from '@emotion/react';
+import { body, space } from '@guardian/source-foundations';
+import { Link } from '@guardian/source-react-components';
+import type { DCRContainerPalette } from 'src/types/front';
+import type { TrailType } from 'src/types/trails';
+import { generateImageURL } from '../../lib/generateImageURL';
+import { decideContainerOverrides } from '../lib/decideContainerOverrides';
+
+const imageStyles = css`
+	width: 7.5rem;
```
Comment (AshCorr) [inline dotcom-rendering/src/web/components/MiniCard.tsx:10]: should use px and space here

File: dotcom-rendering/src/web/components/MiniCard.tsx (hunk 1 above)
Comment (mxdvl) [reply dotcom-rendering/src/web/components/MiniCard.tsx:41]: I would suggest using the pattern used in [`CardPicture`](https://github.com/guardian/dotcom-rendering/blob/c828e7e9375b1d4ae651fe5b19daae74dc37535a/dotcom-rendering/src/web/components/CardPicture.tsx) and soon [`Avatar`](https://github.com/guardian/dotcom-rendering/blob/b69ffbff6446e7beb30b8e2333b7760566c82618/dotcom-rendering/src/web/components/Avatar.tsx) by improve this with an actual picture tag.

This is not a blocker for this work, however, and can be improved upon at a later stage.

File: dotcom-rendering/src/lib/generateImageURL.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
+/**
+ * Generates a URL for calling the Fastly Image Optimiser.
+ *
+ * @see https://developer.fastly.com/reference/io/
+ * @see https://github.com/guardian/fastly-image-service/blob/main/fastly-io_guim_co_uk/src/main/resources/varnish/main.vcl
+ *
+ */
+export const generateImageURL = ({
```
Comment (mxdvl) [inline dotcom-rendering/src/lib/generateImageURL.ts:8]: Not sure what the benefit of extracting this to a lib versus exporting from `Picture` directly. I would also suggest only using `generateSources`.

File: dotcom-rendering/src/web/components/MiniCard.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+type Props = {
+	trail: TrailType;
+	showImage: boolean;
+	containerPalette?: DCRContainerPalette;
+};
+
+/**
+ * A simpler form of card that does not have any of the complexities of a normal card such as sublinks, kickers, format, etc.
+ *
+ * Primarily used for rendering Snaps in containers such as @see NavList
```
Comment (mxdvl) [inline dotcom-rendering/src/web/components/MiniCard.tsx:35]: I think `@linkcode` will work better

```suggestion
 * Primarily used for rendering Snaps in containers such as {@linkcode NavList}
```

File: dotcom-rendering/src/web/components/NavList.tsx [hunk 5]
```diff
@@ -52,29 +47,42 @@ const liStyles = (containerOverrides?: ContainerOverrides) => css`
...
 
-export const NavList = ({ trails, containerPalette }: Props) => {
+const textColumns = css`
+	${until.tablet} {
+		column-count: 1;
+	}
+	${between.tablet.and.desktop} {
+		column-count: 3;
+	}
+	column-count: 4;
```
Comment (mxdvl) [inline dotcom-rendering/src/web/components/NavList.tsx:57]: We tend to work mobile-first, so we should probably switch this around and go for:
```css
column-count: 1;
${from.tablet} {
  column-count: 3;
}
${from.desktop} {
  column-count: 4;
}
```

File: dotcom-rendering/src/web/components/NavList.tsx [hunk 6]
```diff
@@ -52,29 +47,42 @@ const liStyles = (containerOverrides?: ContainerOverrides) => css`
...
+`;
+
+const mediaColumns = css`
+	${until.tablet} {
+		column-count: 1;
+	}
+	column-count: 3;
+`;
```
Comment (mxdvl) [inline dotcom-rendering/src/web/components/NavList.tsx:65]: Same as above, could we start mobile-first?

**Review by mxdvl — APPROVED** [review]
Nice work 👏 

I’d be tempted to leave the `generateImageURL` method local if at all possible. This is the approach we’ve taken in [`CardPicture`](https://github.com/guardian/dotcom-rendering/pull/6065) and [`Avatar`](https://github.com/guardian/dotcom-rendering/pull/6672)

File: dotcom-rendering/src/web/components/MiniCard.tsx [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+	image: string;
+	alt: string;
+};
+
+const MiniCardPicture = ({ image, alt }: MiniCardPictureProps) => {
+	const sources = generateSources(image, [
+		{ breakpoint: breakpoints.desktop, width: 120 },
+	]);
+
+	const [source] = sources.slice(-1);
```
Comment (mxdvl) [inline dotcom-rendering/src/web/components/MiniCard.tsx:37]: Maybe this is even quicker?

```suggestion
	const [source] = generateSources(image, [
		{ breakpoint: breakpoints.desktop, width: 120 },
	]);
```

File: dotcom-rendering/src/web/components/MiniCard.tsx [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+		containerPalette && decideContainerOverrides(containerPalette);
+
+	return (
+		<Link
+			href={trail.url}
+			priority="secondary"
+			subdued={true}
+			cssOverrides={[linkStyles, linkOverrideStyles(containerOverrides)]}
+		>
+			{showImage && trail.image && (
```
Comment (mxdvl) [inline dotcom-rendering/src/web/components/MiniCard.tsx:70]:
```suggestion
			{showImage && !!trail.image && (
```

**Review by mxdvl — APPROVED** [review]

**Pull request merged**